### PR TITLE
fix(documentarray): fix save binary

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
Background:
https://github.com/jina-ai/jina/issues/2540

### CODE REVIEW

File: jina/types/arrays/document.py [hunk 1]
```diff
@@ -307,7 +307,13 @@ def save_binary(self, file: Union[str, BinaryIO]) -> None:
 
         with file_ctx as fp:
             dap = DocumentArrayProto()
-            dap.docs.extend([d.proto for d in self._docs_proto])
+            if len(self._docs_proto) > 0:
+                if type(self._docs_proto[0]) != DocumentProto:
```
Comment (hanxiao) [inline jina/types/arrays/document.py:311]: use `is_instance`

File: jina/types/arrays/document.py [hunk 2]
```diff
@@ -307,7 +307,13 @@ def save_binary(self, file: Union[str, BinaryIO]) -> None:
 
         with file_ctx as fp:
             dap = DocumentArrayProto()
-            dap.docs.extend([d.proto for d in self._docs_proto])
+            if len(self._docs_proto) > 0:
```
Comment (hanxiao) [inline jina/types/arrays/document.py:310]:
```suggestion
            if self._docs_proto:
```

File: jina/types/arrays/document.py [hunk 3]
```diff
@@ -307,7 +307,13 @@ def save_binary(self, file: Union[str, BinaryIO]) -> None:
...
+                if type(self._docs_proto[0]) != DocumentProto:
+                    dap.docs.extend([d.proto for d in self._docs_proto])
+                else:
+                    dap.docs.extend([d for d in self._docs_proto])
+            else:
+                default_logger.warning('data is empty')
```
Comment (hanxiao) [inline jina/types/arrays/document.py:316]: do not raise error or warning, empty content is a valid content in IO ops

**Review by hanxiao — CHANGES_REQUESTED** [review]
please change according to the reviews

File: jina/types/arrays/document.py (hunk 3 above)
Comment (Yongxuanzhang) [reply jina/types/arrays/document.py:316]: Sure

File: jina/types/arrays/document.py [hunk 4]
```diff
@@ -307,7 +307,11 @@ def save_binary(self, file: Union[str, BinaryIO]) -> None:
 
         with file_ctx as fp:
             dap = DocumentArrayProto()
-            dap.docs.extend([d.proto for d in self._docs_proto])
+            if self._docs_proto:
+                if isinstance(self._docs_proto[0], DocumentProto):
+                    dap.docs.extend([d for d in self._docs_proto])
```
Comment (hanxiao) [inline jina/types/arrays/document.py:312]:
```suggestion
                    dap.docs.extend(self._docs_proto)
```

**Pull request merged**